### PR TITLE
fix: batch channel not close when client close

### LINKED ISSUE #100: for select usage error
https://github.com/openGemini/opengemini-client-go/blob/0a8ed4306924dcc311acda5517b0c920ea4a00a1/opengemini/write.go#L128-L149

Same error for ` for select`. Donot do function after select. It will be stack when signal income. And the behavior in next loop is unforeseeable

### PR DESCRIPTION
**Pull request by cyruslo**
Fix #100

### CODE REVIEW

**Review by shoothzj — APPROVED** [review]

**Comment by xuthus5** [comment]
the client will block because the channel is not closed, a simple demo to verify the idea, `End of data` will never have a chance to print.
```go
package main

import (
	"fmt"
	"time"
)

func reader(ch <-chan int) {
	for record := range ch {
		fmt.Println(record)
	}
	fmt.Println("End of data")
}

func main() {
	var ch = make(chan int)
	go reader(ch)
	for i := 0; i < 10; i++ {
		ch <- i
	}
	time.Sleep(time.Second * 10)
}
```

**Comment by shoothzj** [comment]
Currently, we don't have any restrictions, so we can still call `WritePointWithRp` after closing it. It may contains some leak.

**Comment by xuthus5** [comment]
> Currently, we don't have any restrictions, so we can still call after closing it. It may contains some leak.`WritePointWithRp`

`Close()` semantics have no effect.

File: opengemini/write.go [hunk 1]
```diff
@@ -27,6 +27,11 @@ type sendBatchWithCB struct {
 	callback WriteCallback
 }
 
+func (s *sendBatchWithCB) Clear() {
```
Comment (xuthus5) [inline opengemini/write.go:30]:
```suggestion
func (s *sendBatchWithCB) clear() {
```

File: opengemini/write.go [hunk 2]
```diff
@@ -27,6 +27,11 @@ type sendBatchWithCB struct {
...
+	s.point = nil
+	s.callback(fmt.Errorf("sendBatchWithCB timeout"))
```
Comment (xuthus5) [inline opengemini/write.go:32]:
```suggestion
	s.callback(fmt.Errorf("send batch context cancelled"))
```

File: opengemini/write.go [hunk 3]
```diff
@@ -129,6 +134,12 @@ func (c *client) internalBatchSend(ctx context.Context, database string, rp stri
 		select {
 		case <-ctx.Done():
 			ticker.Stop()
+			for record := range resource {
+				record.Clear()
+				if len(resource) == 0 {
```
Comment (xuthus5) [inline opengemini/write.go:139]: the `resource` is closed from the `Close` method.

**Comment by xuthus5** [comment]
Close method:
```go
func (c *client) Close() error {
	c.batchContextCancel()
	c.dataChanMap.Range(func(key, value interface{}) bool {
		close(value.(chan *sendBatchWithCB))
		c.dataChanMap.Delete(key)
		return true
	})
	return nil
}
```

WritePointWithRp method line 57, add a context cancel judgment to prevent the program from writing data to a closed channel:
```go
if c.batchContext.Err() != nil {
    return errors.New("batch write failed, error: " + c.batchContext.Err().Error())
}
```

**Pull request merged**